files-community/Files
# Feature: Added a prompt when failing to rename items requiring additional permissions

**Resolved / Related Issues**
   Closes #8326 

**Known bugs**
- There are two cases where `Edit permissions` doesn't work:
  1. You start to rename an item on one page, and then (somehow) navigate to another before clicking `Edit permissions`
  2. You open the properties window for that item, navigate to a different page and then rename it

The issue behind these bugs is that the `Properties` window depends on `IShellPage` and `ListedItem`, while `RenameAsync` doesn't. To get those values, I'm using `ContentPageContext`, but this doesn't work if you change the working directory while renaming. Do you have any ideas on how I can fix this without messing up the code too much?

**Validation**
How did you test these changes?
   1. Navigate to `C:\Windows\Ssytem32`
   4. Try to rename any file/folder (such as `Utilman.exe`)
   5. When the warning is shown try clicking both `Ok` and `Edit permissions`

**Screenshots**
![Screenshot 2024-02-07 210409](https://github.com/files-community/Files/assets/102259289/21ed996c-96f6-4c88-85dc-48d026600096)
